src/puyapy/awst_build/eb/arc4/_base.py

```diff
@@ -138,9 +138,8 @@ class _ARC4ArrayExpressionBuilder(BytesBackedInstanceExpressionBuilder[pytypes.A
     @typing.override
     def iterate(self) -> Expression:
         if not self.pytype.items.wtype.immutable:
-            logger.error(
-                "cannot directly iterate an ARC4 array of mutable objects,"
-                " construct a for-loop over the indexes via urange(<array>.length) instead",
+            logger.info(
+                "use `algopy.urange(<array>.length)` to iterate by index",
```

Comment:
why remove the fact that it's due to mutable values from the error message here?

Reply:
oh I see - maybe add a code comment that this condition creates an error during validation in puya core

Reply:
Added comment